plugin/pkg/admission/securitycontext/scdeny/admission.go

```diff
@@ -53,42 +60,57 @@ func (p *plugin) Admit(a admission.Attributes) (err error) {
 		return apierrors.NewBadRequest("Resource was marked with kind Pod but was unable to be converted")
 	}
 
-	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SupplementalGroups != nil {
```

Comment:
I feel that you have extracted too much. I'm expecting that the logic of iteration will stay on the same place, but only logic for checking `SecurityContext`/`PodSecurityContext` will be extracted.

Something like this:
```go
if err := checkPodSecurityContext(pod.Spec.SecurityContext); err != nil {
    return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
}

for _, container := range pod.Spec.InitContainers {
    if err := checkSecurityContext(container.SecurityContext); err != nil {
        return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
    }
}

for _, container := range pod.Spec.Containers {
    if err := checkSecurityContext(container.SecurityContext); err != nil {
        return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
    }
}
```